cpp/src/reductions/reduction_operators.cuh

```diff
@@ -1,58 +1,183 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef CUDF_REDUCTION_OPERATORS_CUH
+#define CUDF_REDUCTION_OPERATORS_CUH
+
 #include <cudf/cudf.h>
 #include <utilities/cudf_utils.h>
 #include <utilities/wrapper_types.hpp>
 #include <utilities/error_utils.hpp>
+#include <iterator/iterator.cuh>
+
+#include <utilities/device_operators.cuh>
 
-#include <utilities/device_atomics.cuh>
+#include <cmath>
 
 namespace cudf {
 namespace reductions {
 
-// ------------------------------------------------------------------------
-// difinitions of device struct for binary operation
+template<typename T>
+struct meanvar_no_count
```

Comment:
Documentation?  What does `no_count` mean?

Reply:
I will change the name into `var_std` since it is used to compute `var` and `std`.
I thought it to differentiate to `cudf::meanvar` https://github.com/rapidsai/cudf/blob/branch-0.9/cpp/src/iterator/transform_unary_functions.cuh#L50